app/code/core/Mage/Adminhtml/controllers/Tax/RuleController.php

```diff
@@ -110,7 +110,8 @@ public function saveAction()
             return $this->getResponse()->setRedirect($this->getUrl('*/tax_rule'));
         }
 
-        $ruleModel = $this->_getSingletonModel('tax/calculation_rule');
+        $ruleId = (int)$this->getRequest()->getParam('tax_calculation_rule_id');
+        $ruleModel = $this->_getSingletonModel('tax/calculation_rule')->load($ruleId);
```

Comment:
Using a singleton here to load by ID seems a bit dangerous. Is the singleton needed here?
Not an argument against the PR, as its happening limited to this PR, and a Singleton was already used before the change.
But wanted to at least note it.